src/GraphQL.Tests/CoreToVanillaConverterTests.cs

```diff
@@ -64,10 +65,23 @@ public async Task DocumentExecuter_really_small_double_valid()
         [InlineData(ASTNodeKind.BooleanValue, "false")]
         public void Values_Parse_Successfully(ASTNodeKind kind, string valueString)
         {
+            ASTNode BuildNode()
+            {
+                return kind switch
+                {
+                    ASTNodeKind.IntValue => new GraphQLIntValue(),
+                    ASTNodeKind.FloatValue => new GraphQLFloatValue(),
+                    ASTNodeKind.BooleanValue => new GraphQLBooleanValue(),
+                    _ => throw new NotSupportedException(),
+                };
+            }
+
             //note: thousand separators and/or culture-specific characters are invalid graphql literals, and will not be returned by graphql-parser
             //uppercase TRUE and FALSE are also invalid graphql input data, and will not be returned by graphql-parser
             //whitespace will not be returned by graphql-parser
-            _ = CoreToVanillaConverter.Value(new GraphQLScalarValue(kind) { Value = valueString });
+            dynamic node = BuildNode();
```

Comment:
🤔 I may add interface in parser. Something like `IHasValueNode`. But I doubt.

Reply:
If this only is an issue here in this test, it's okay...   I was thinking about this issue in my review of the parser, I think...